### ISSUE TITLE
Documentation build complains a lot

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#4824]
The docs build was in a reasonable state a release or two back, but now the warnings are stacking up.

Tasks:
* [x] Fix documentation build so that it fails the build, forcing developers to fix them.
* [x] Fix the current warnings, example below.

```
$ make V=0 --quiet -C Documentation/ dummy SPHINXOPTS="-q" 2>&1
  CHECK documentation dependencies...
  CHECK documentation spelling...
/home/joe/work/src/github.com/cilium/cilium/Documentation/api.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/bpf.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/cheatsheet.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/cmdref/cli_index.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/cmdref/index.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/cmdref/kvstore.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/commit-access.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/concepts.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/configuration/metrics.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/contributing.rst:5: WARNING: Unexpected indentation.
WARNING: /home/joe/work/src/github.com/cilium/cilium/Documentation/contributing.rst:29: (ERROR/3) Malformed table.

+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
| Dependency                                                                       | Version / Commit ID      | Download Command                                                              |
+==================================================================================+==========================+===============================================================================+
| git                                                                              | latest                   | N/A (OS-specific)                                                             |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
| `go <https://golang.org/dl/>`_                                                   | 1.10                      | N/A (OS-specific)                                                             |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
| `dep <https://github.com/golang/dep/>`_                                          | >= v0.4.1                | ``curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh``  |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
| `go-bindata <https://github.com/cilium/go-bindata>`_                             | ``a0ff2567cfb``          | ``go get -u github.com/cilium/go-bindata/...``                                |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
+ `ginkgo <https://github.com/onsi/ginkgo>`__                                      | >= 1.4.0                 | ``go get -u github.com/onsi/ginkgo``                                          |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
+ `gomega <https://github.com/onsi/gomega>`_                                       | >= 1.2.0                 | ``go get -u github.com/onsi/gomega``                                          |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
| `protoc-gen-go <https://github.com/golang/protobuf/tree/master/protoc-gen-go>`_  | latest                   | ``go get -u github.com/golang/protobuf/protoc-gen-go``                        |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
| `protoc-gen-validate <https://github.com/lyft/protoc-gen-validate>`_             | latest                   | ``go get -u github.com/lyft/protoc-gen-validate``                             |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
+ `Docker <https://docs.docker.com/engine/installation/>`_                         | OS-Dependent             | N/A (OS-specific)                                                             |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
+ `Docker-Compose <https://docs.docker.com/compose/install/>`_                     | OS-Dependent             | N/A (OS-specific)                                                             |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
+ `Cmake  <https://cmake.org/download/>`_                                          | OS-Dependent             | N/A (OS-specific)                                                             |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
+ `Bazel <https://docs.bazel.build/versions/master/install.html>`_                 | 0.14.1                   | N/A (OS-specific)                                                             |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
+ `Libtool <https://www.gnu.org/software/libtool/>`_                               | >= 1.4.2                 | N/A (OS-specific)                                                             |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
+ `Automake <https://www.gnu.org/software/automake/>`_                             | OS-Dependent             | N/A (OS-specific)                                                             |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
+ `Kubecfg <https://github.com/ksonnet/kubecfg>`_                                  | >=0.8.0                  | go get github.com/ksonnet/kubecfg                                             |
+----------------------------------------------------------------------------------+--------------------------+-------------------------------------------------------------------------------+
/home/joe/work/src/github.com/cilium/cilium/Documentation/contributing.rst:10: WARNING: Duplicate explicit target name: "jenkins".
/home/joe/work/src/github.com/cilium/cilium/Documentation/docker/index.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/further_reading.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettinghelp.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/cilium_install.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/cilium_install_crio.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/cilium_install_docker.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/docker.rst:5: WARNING: Unexpected indentation.
gettingstarted/gsg_intro.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/elasticsearch.rst:5: WARNING: Unexpected indentation.
gettingstarted/gsg_intro.rst:5: WARNING: Unexpected indentation.
gettingstarted/minikube_intro.rst:5: WARNING: Unexpected indentation.
gettingstarted/cilium_install.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/grpc.rst:5: WARNING: Unexpected indentation.
gettingstarted/gsg_intro.rst:5: WARNING: Unexpected indentation.
gettingstarted/minikube_intro.rst:5: WARNING: Unexpected indentation.
gettingstarted/cilium_install.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/gsg_intro.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/gsg_starwars.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/index.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/istio.rst:5: WARNING: Unexpected indentation.
gettingstarted/gsg_intro.rst:5: WARNING: Unexpected indentation.
gettingstarted/minikube_intro.rst:5: WARNING: Unexpected indentation.
gettingstarted/cilium_install.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/kafka.rst:5: WARNING: Unexpected indentation.
gettingstarted/gsg_intro.rst:5: WARNING: Unexpected indentation.
gettingstarted/minikube_intro.rst:5: WARNING: Unexpected indentation.
gettingstarted/cilium_install.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/mesos.rst:5: WARNING: Unexpected indentation.
gettingstarted/gsg_intro.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/minikube.rst:5: WARNING: Unexpected indentation.
gettingstarted/gsg_intro.rst:5: WARNING: Unexpected indentation.
gettingstarted/minikube_intro.rst:5: WARNING: Unexpected indentation.
gettingstarted/cilium_install.rst:5: WARNING: Unexpected indentation.
gettingstarted/gsg_starwars.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/minikube_intro.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/glossary.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/index.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/advanced_options.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/from_source.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/index.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kops.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kubernetes.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/system_requirements.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/upgrade.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/upgrade.rst:11: WARNING: Duplicate explicit target name: "github pr".
/home/joe/work/src/github.com/cilium/cilium/Documentation/intro.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/istio/index.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/kubernetes/ciliumendpoint.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/kubernetes/compatibility.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/kubernetes/index.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/kubernetes/install.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/kubernetes/intro.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/kubernetes/policy.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/kubernetes/quickinstall.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/kubernetes/troubleshooting.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/mesos/index.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/policy/index.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/policy/intro.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/policy/language.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/policy/lifecycle.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/policy/troubleshooting.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/troubleshooting.rst:5: WARNING: Unexpected indentation.
/home/joe/work/src/github.com/cilium/cilium/Documentation/troubleshooting.rst:444: WARNING: Cannot analyze code. No Pygments lexer found for "bash:".
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/cilium_install_crio.rst: WARNING: document isn't included in any toctree
/home/joe/work/src/github.com/cilium/cilium/Documentation/gettingstarted/cilium_install_docker.rst: WARNING: document isn't included in any toctree
/home/joe/work/src/github.com/cilium/cilium/Documentation/kubernetes/compatibility.rst: WARNING: document isn't included in any toctree
/home/joe/work/src/github.com/cilium/cilium/Documentation/contributing.rst:1122: WARNING: undefined label: trigger_phrases` (if the link has no caption the label must precede a section header)
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kops.rst:97: WARNING: 'any' reference target not found: com-company-emailid-
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kops.rst:138: WARNING: 'any' reference target not found: kops update cluster
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kube-router.rst:111: WARNING: 'any' reference target not found: 10.2.0.0/24 via 10.2.0.172 dev cilium_host src 10.2.0.172
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kube-router.rst:114: WARNING: 'any' reference target not found: 10.2.1.0/24 via 172.0.51.175 dev eth0 proto 17
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kube-router.rst:119: WARNING: 'any' reference target not found: 10.2.2.0/24 dev tun-172011760 proto 17 src 172.0.50.227
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kube-router.rst:119: WARNING: 'any' reference target not found: 10.2.3.0/24 dev tun-1720186231 proto 17 src 172.0.50.227
/home/joe/work/src/github.com/cilium/cilium/Documentation/policy/language.rst:48: WARNING: undefined label: k8s_namespace (if the link has no caption the label must precede a section header)
/home/joe/work/src/github.com/cilium/cilium/Documentation/troubleshooting.rst:98: WARNING: 'any' reference target not found: cilium-1234
/home/joe/work/src/github.com/cilium/cilium/Documentation/troubleshooting.rst:98: WARNING: 'any' reference target not found: kubectl -n kube-system get pods -l k8s-app=cilium
/home/joe/work/src/github.com/cilium/cilium/Documentation/troubleshooting.rst:180: WARNING: 'any' reference target not found: kubectl exec
WARNING: Not copying tabs assets! Not compatible with spelling builder
  GEN   Documentation/_build/dummy
/home/joe/work/src/github.com/cilium/cilium/Documentation/contributing.rst:1122: WARNING: undefined label: trigger_phrases` (if the link has no caption the label must precede a section header)
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kops.rst:97: WARNING: 'any' reference target not found: com-company-emailid-
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kops.rst:138: WARNING: 'any' reference target not found: kops update cluster
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kube-router.rst:111: WARNING: 'any' reference target not found: 10.2.0.0/24 via 10.2.0.172 dev cilium_host src 10.2.0.172
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kube-router.rst:114: WARNING: 'any' reference target not found: 10.2.1.0/24 via 172.0.51.175 dev eth0 proto 17
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kube-router.rst:119: WARNING: 'any' reference target not found: 10.2.2.0/24 dev tun-172011760 proto 17 src 172.0.50.227
/home/joe/work/src/github.com/cilium/cilium/Documentation/install/guides/kube-router.rst:119: WARNING: 'any' reference target not found: 10.2.3.0/24 dev tun-1720186231 proto 17 src 172.0.50.227
/home/joe/work/src/github.com/cilium/cilium/Documentation/policy/language.rst:48: WARNING: undefined label: k8s_namespace (if the link has no caption the label must precede a section header)
/home/joe/work/src/github.com/cilium/cilium/Documentation/troubleshooting.rst:98: WARNING: 'any' reference target not found: cilium-1234
/home/joe/work/src/github.com/cilium/cilium/Documentation/troubleshooting.rst:98: WARNING: 'any' reference target not found: kubectl -n kube-system get pods -l k8s-app=cilium
/home/joe/work/src/github.com/cilium/cilium/Documentation/troubleshooting.rst:180: WARNING: 'any' reference target not found: kubectl exec
WARNING: Not copying tabs assets! Not compatible with dummy builder
```

**Comment by ishantanu** [comment]
@joestringer Since the lines mentioned in the description are flagged as warnings, do you mean to add `-W` to sphinx-build? So that the build will fail if there are some warnings? Because there are a lot of "mistakes" which are being flagged as `WARNING` and they need to be fixed before the build succeeds.

**Comment by eloycoto** [comment]
@joestringer yesterday I made a few cleanups, not all but a few of them: 
https://github.com/cilium/cilium/pull/4813

**Comment by joestringer** [comment]
@iamshantanu maybe? Except I think that the dummy docs build triggered from the main Makefile attempts to filter out known issues like the dummy backend doesn't support one of the plugins we use for tabbed sections (eg k8s install instructions for different k8s versions). This could potentially hide a failure in the Documentation/ Makefile if we're not careful.

And yes, all existing issues would need to be fixed before turning something like that on.

**Comment by ishantanu** [comment]
@joestringer Understood. Also, I guess you meant "@iamShantanu101" instead of "@iamshantanu".

**Comment by joestringer** [comment]
Unfortunately it looks like the docs build would always fail if we failed on warnings due to this message:

```
WARNING: Not copying tabs assets! Not compatible with dummy builder
```

However, perhaps we could tweak the documentation target so that once it greps out the above error, it searches the text for `WARNING` and fails if there is any match?

**Comment by joestringer** [comment]
Fixed via #5591.
